Makefile

```
@@ -45,3 +44,26 @@ tidy::
 	cd pkg/tests && go mod tidy
 	cd testing && go mod tidy
 	go mod tidy
+
+# Update the subtree in pkg/tf2pulumi/internal/tf with the latest code from terraform master
+TFPKGS=backend\|command\|provider-terraform\|provider-simple\|provider-simple-v6\|provisioner-local-exec\|cloud\|builtin\|grpcwrap\|initwd\|earlyconfig\|moduledeps\|plugin\|tfplugin5\|plugin6\|tfplugin6\|communicator\|providercache\|e2e\|legacy\|moduletest\|helper\|repl\|terminal
+pull_terraform::
+	git remote remove terraform || true
+	git remote add -f -t main --no-tags terraform https://github.com/hashicorp/terraform.git
+	git rm -rf ./pkg/tf2pulumi/internal/tf
+	git read-tree --prefix=pkg/tf2pulumi/internal/tf/ -u terraform/main:internal
+	# Remove the parts of Terraform we don't need
+	find ./pkg/tf2pulumi/internal/tf -mindepth 1 -maxdepth 1 -type d -regex ".*/tf/\($(TFPKGS)\)" -exec rm -rf {} \;
+	# No need for us to keep any of Terraforms tests
+	find ./pkg/tf2pulumi/internal/tf -type f -regex ".*_test.go" -delete
+	# Update the .go file import paths
+	find ./pkg/tf2pulumi/internal/tf -type f -name "*.go" -exec sed -i -e 's|github.com/hashicorp/terraform/internal|github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tf2pulumi/internal/tf|g' {} \;
+	# Workaround the duplicate registration of rpcFriendlyDiag in tf/tfdiags/rpc_friendly.go and same type in the public plugin repo
+	sed -i -e 's|rpcFriendlyDiag|internalRpcFriendlyDiag|g' ./pkg/tf2pulumi/internal/tf/tfdiags/rpc_friendly.go
```

If you end up using perl:
```suggestion
	perl -p -i -e 's|rpcFriendlyDiag|internalRpcFriendlyDiag|g' ./pkg/tf2pulumi/internal/tf/tfdiags/rpc_friendly.go
```